adding function add_columns_each_beach() plus a few other edits

beckeroobonsai opened the pull request:
- Added function add_columns_each_beach() that adds to the dataframe the previous day's ecoli reading from other beaches, each in a separate column.  See issue #90 
- Made changes to read_data_simplified() so that it cleans up beaches and deletes weekend readings.
- Made read_data() function have an option for deleting the weekend readings.
- Edited group_beaches_geographically() so that Leone beach is only in group 2, not in both group 1 and 2.
- Edited add_column_prior_data() function so that it no longer returns rows for days where there was no reading.

beckeroobonsai committed:
Update read_data.py

Added functions:

    addColumnPriorData(df, colName , NdaysPrior):
        Returns a new dataframe with an additional column of data N days Prior to current.

    cleanUpBeaches(data):
        Makes the beachnames clean up a separate function that also checks for 
        duplicate readings at a given beach for a given day and takes the higher reading. 

    read_data_simplified():
        Simplifies data reading (no more need for split_sheets or date_lookup functions).
        Also sorts Reading1 and Reading2 into lowReading and highReading columns.
        Does minimal cleaning of data. 
        No merging onto Drek or weather data.

beckeroobonsai committed:
Add files via upload

- cleaned up cleanbeachnames2.csv
- debugged the function in read_data for adding column of prior day's data
- script for building a GBM model

beckeroobonsai committed:
Add files via upload

added precision_recall_curve to script

beckeroobonsai committed:
Added files via upload

beckeroobonsai committed:
added Drek names

beckeroobonsai committed:
Add files via upload

beckeroobonsai committed:
Update read_data.py

- Added function add_columns_each_beach() that adds to the dataframe the previous day's ecoli reading from other beaches, each in a separate column.  
- Made changes to read_data_simplified() so that it cleans up beaches and deletes weekend readings.
- Made read_data() function have an option for deleting the weekend readings.
- Edited group_beaches_geographically() so that Leone beach is only in group 2, not in both group 1 and 2.
- Edited add_column_prior_data() function so that it no longer returns rows for days where there was no reading.

beckeroobonsai committed:
Update read_data.py

added "import numpy as np" and changed line in add_columns_each_beach() function to use np.isnan() rather than isnan()

beckeroobonsai commented:
noticed an error when I ran the new function add_columns_each_beach().  
It needs isnan() from numpy.
Put in the header  "import numpy as np" and change isnan() to np.isnan()

I'm not sure how to update the file after creating a pull request.

kbrose commented:
> * Made read_data() function have an option for deleting the weekend readings.

I would rather keep the weekend readings when judging performance, but remove them from the predictors. I am planning on adding an issue to track this problem. If it's just an option, though, probably not a big deal.

> * Edited add_column_prior_data() function so that it no longer returns rows for days where there was no reading.

Don't know if I understand this exactly - what if a row has other information in it? E.g., pulling the 1-day previous E. coli reading on Monday (so that you are trying to get Sunday's reading) might return a `nan`. Do you delete that whole row even though everything else is filled in?

ChrisProkop commented:
Thanks for catching my doubling up on Leone.

beckeroobonsai committed:
cleaned dataset to use for modeling

beckeroobonsai committed:
cleaned dataset to use for modeling

This has prior readings from all beaches merged on. As well as mean of each beach group. 
There are three days of historical data for 
'precipIntensity', 'precipIntensityMax', 'apparentTemperatureMax','apparentTemperatureMaxTime',
                'windSpeed', 'windBearing', 'visibility', 'cloudCover','precipIntensityMaxTime'
This also includes hourly data for all those weather variables.

beckeroobonsai committed:
pickled models

beckeroobonsai committed:
Delete RFclassifSouth_2014.pkl

beckeroobonsai committed:
Delete RFclassifSouth_2013.pkl
